model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -215,17 +215,21 @@ message Metrics {
   message PTransform {
     // Metrics that are measured for processed and active element groups.
     message Measured {
-      // (Required) Map from local input name to number of elements processed
+      // (Optional) Map from local input name to number of elements processed
       // from this input.
+      // If unset, assumed to be the sum of the outputs of all producers to
```

Comment:
Since Measured is used in both ProcessedElements and ActiveElements, what does this comment mean since I don't believe you could sum the outputs of ActiveElements from upstream to get the current number of ActiveElements in this PTransform.

Reply:
Good point. Expanded comment. (Could delete this comment entirely if you'd prefer.)

Reply:
I think we would want this information for writing to the data channel on the output side as well since this would tell the runner harness that there is a lot buffered between what the SDK harness has "transmitted" and what the Runner harness has consumed?

Is there a way to clarify this comment to not confuse people so that they don't put arbitrary data in here for things like sources and sinks? Alternatively we could have a dedicated map for the data channels that say how many bytes they have read/written or input/output element counts.

Reply:
Yes, we want information about the SDK's view of how much data has been read from/written to the channel. 

For now I'm not even going to populate it for the grpc operations; the system injects these IO operations and knows there's a 1:1 relationship between their inputs and outputs. 

Sources and sinks can't get access to these counters to populate them.